rascal-vscode-extension/src/RascalExtension.ts

```diff
@@ -148,6 +147,49 @@ export class RascalExtension implements vscode.Disposable {
         }
     }
 
+    private getTerminalOrigin(uri: vscode.Uri | undefined, extraArgs: string[]): string {
+        if (uri) {
+            const config = vscode.workspace.getConfiguration();
+            const originFormat = config.get('rascal.terminal.name.originFormat');
+            switch (originFormat) {
+                case 'Project root': {
+                    const projectRoot = vscode.workspace.getWorkspaceFolder(uri);
+                    if (projectRoot && projectRoot.name) {
+                        return projectRoot.name;
+                    }
+                    return "no project";
+                }
+                case 'Module (qualified)': {
+                    if (extraArgs[0] === '--loadModule' &&
+                        extraArgs[1] && extraArgs[1].match(this.qualifiedName)) {
+                        return extraArgs[1];
+                    }
+                    return "no module";
+                }
+                case 'Module (unqualified)': {
+                    if (extraArgs[0] === '--loadModule' && extraArgs[1]) {
+                        const name = extraArgs[1].match(this.qualifiedName);
+                        if (name && name[1]) {
+                            return name[1];
+                        }
+                    }
+                    return "no module";
+                }
+                default:
+                    console.log(`Unknown origin format: ${originFormat}`);
+            }
+        }
+        return 'no project or module';
+    }
+
+    private qualifiedName: RegExp = (() => {
+        const name1 = '(?:[A-Z_a-z][0-9A-Z_a-z]*)';
+        const name2 = '(?:\\\\[A-Z_a-z][\\-0-9A-Z_a-z]*)';
+        const name = `(?:${name1}|${name2})`;
+        const qualifiedName = `(?:(?:${name}::)*(${name}))`;
+        return new RegExp(`^${qualifiedName}$`);
+    })(); // Build the regex only once
```

Comment:
If I remember correctly, a JavaScript regex cannot be reused.

But if you care for performance, a regex literal is a lot faster. I think since the jitter can decide to reuse the complication result.

Reply:
> If I remember correctly, a JavaScript regex cannot be reused.

Only when the `global`/`sticky` flag is set ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test))

Reply:
aah right :)